test/abstractarray.jl

```diff
@@ -840,3 +840,9 @@ end
 @testset "checkbounds_indices method ambiguities #20989" begin
     @test Base.checkbounds_indices(Bool, (1:1,), ([CartesianIndex(1)],))
 end
+
+@testset "ImageCore #40" begin
+    Base.convert(::Type{Array{T,n}}, a::Array) where {T<:Number,n} =
```

Comment:
(Also presumably `dest` is supposed to be `Dict(:a=>1, :b=>2.0)` as in the linked issue)